vizro-core/src/vizro/actions/_action_loop/_build_action_loop_callbacks.py

```diff
@@ -1,31 +1,30 @@
 """Contains utilities to create required dash callbacks for the action loop."""
-import json
 import logging
-from typing import Any, Dict, List, Optional
+from typing import List
 
-from dash import Input, Output, State, callback, clientside_callback, ctx, dcc, no_update
-from dash.exceptions import PreventUpdate
+from dash import ClientsideFunction, Input, Output, State, clientside_callback
 
-from vizro._constants import ON_PAGE_LOAD_ACTION_PREFIX
 from vizro.actions._action_loop._action_loop_utils import (
     _get_actions_chains_on_registered_pages,
     _get_actions_on_registered_pages,
 )
-from vizro.managers import model_manager
 
 logger = logging.getLogger(__name__)
 
 
 def _build_action_loop_callbacks() -> None:
     """Creates all required dash callbacks for the action loop."""
-    # TODO - Reduce the number of the callbacks in the action loop mechanism
     actions_chains = _get_actions_chains_on_registered_pages()
     actions = _get_actions_on_registered_pages()
 
-    gateway_triggers: List[Input] = []
+    if not actions_chains:
+        return
```

Comment:
(for later) Why has that appeared?

Reply:
This prevents the creation of action loop callbacks if there are no actions defined in the system. It only covers the edge case if users create a dashboard without charts, filters, parameters...